File: 274-h-index/h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        # easy approach: less optimal O(n log n)
        # will improve to O(n) using extra spac
        
        citations.sort(reverse=True)
        
        for i in range(len(citations)):
            papers = i + 1
            if papers > citations[i]:
                return papers - 1
            if papers == citations[i]:
                return papers
        return papers

File: 274-h-index/h-index.swift
class Solution {
    func hIndex(_ citations: [Int]) -> Int {
        let n: Int = citations.count
        var count: [Int] = Array(repeating: 0, count: n + 1)

        for c in citations {
            if c >= n {
                count[n] += 1
            } else {
                count[c] += 1
            }
        }

        var total: Int = 0
        for i in stride(from: n, through: 0, by: -1) {
            total += count[i]
            if total >= i {
                return i
            }
        }
        return 0
    }
}